src/Nethermind/Nethermind.JsonRpc.Test/Modules/AdminModuleTests.cs

```diff
@@ -0,0 +1,105 @@
+    {
+        private IAdminModule _adminModule;
+        private EthereumJsonSerializer _serializer;
+        private NetworkConfig _networkConfig;
+        private IBlockTree _blockTree;
+        private const string _enodeString = "enode://44826a5d6a55f88a18298bca4773fca5749cdc3a5c9f308aa7d810e9b31123f3e7c5fba0b1d70aac5308426f47df2a128a6747040a3815cc7dd7167d03be320d@127.0.0.1:30303";
+        
+        [SetUp]
+        public void Setup()
+        {
+            _blockTree = Build.A.BlockTree().OfChainLength(5).TestObject;
+            _networkConfig = new NetworkConfig();
+            IPeerManager peerManager = Substitute.For<IPeerManager>();
+            peerManager.ActivePeers.Returns(new List<Peer> {new Peer(new Node("127.0.0.1", 30303, true))});
+            
+            IStaticNodesManager staticNodesManager = Substitute.For<IStaticNodesManager>();
+            Enode enode = new Enode(_enodeString);
+            _adminModule = new AdminModule(_blockTree, _networkConfig, peerManager, staticNodesManager, enode, "C:\\dataDir");
+            _serializer = new EthereumJsonSerializer();
+        }
+        
+        [Test]
+        public void Test_node_info()
+        {
+            string serialized = RpcTest.TestSerializedRequest(_adminModule, "admin_nodeInfo");
+            JsonRpcSuccessResponse response = _serializer.Deserialize<JsonRpcSuccessResponse>(serialized);
+            JsonSerializerSettings settings = new JsonSerializerSettings();
+            settings.Converters = EthereumJsonSerializer.BasicConverters;
+            
+            NodeInfo nodeInfo = ((JObject) response.Result).ToObject<NodeInfo>(JsonSerializer.Create(settings));
+            nodeInfo.Enode.Should().Be(_enodeString);
+            nodeInfo.Id.Should().Be("44826a5d6a55f88a18298bca4773fca5749cdc3a5c9f308aa7d810e9b31123f3e7c5fba0b1d70aac5308426f47df2a128a6747040a3815cc7dd7167d03be320d");
+            nodeInfo.Ip.Should().Be("127.0.0.1");
+            nodeInfo.Name.Should().Be(ClientVersion.Description);
+            nodeInfo.ListenAddress.Should().Be("127.0.0.1:30303");
+            nodeInfo.Ports.Discovery.Should().Be(_networkConfig.DiscoveryPort);
+            nodeInfo.Ports.Listener.Should().Be(_networkConfig.P2PPort);
+
+            nodeInfo.Protocols.Should().HaveCount(1);
+            nodeInfo.Protocols["eth"].Difficulty.Should().Be(_blockTree.Head.TotalDifficulty ?? 0);
+            nodeInfo.Protocols["eth"].HeadHash.Should().Be(_blockTree.HeadHash);
+            nodeInfo.Protocols["eth"].GenesisHash.Should().Be(_blockTree.GenesisHash);
+            nodeInfo.Protocols["eth"].ChainId.Should().Be(_blockTree.ChainId);
+        }
+        
+        [Test]
+        public void Test_data_dir()
+        {
+            string serialized = RpcTest.TestSerializedRequest(_adminModule, "admin_dataDir");
+            JsonRpcSuccessResponse response = _serializer.Deserialize<JsonRpcSuccessResponse>(serialized);
+            response.Result.Should().Be("C:\\dataDir");
```

Comment:
is this some required path?

Reply:
hmmmm, it should be from the InitCOnfig, let me check if I hardcoded it by mistake

Reply:
no, it is fine - it comes from constructor above - let me extract it to some constant